pkg/infra/benchmark_test.go

```diff
@@ -0,0 +1,80 @@
+package infra
+
+import (
+	"net"
+	"testing"
+
+	"github.com/guoger/tape/e2e/mock"
+	"github.com/hyperledger/fabric-protos-go/peer"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+func MockingPeer() (*mock.Server, string) {
+	lis, _ := net.Listen("tcp", "127.0.0.1:0")
+	grpcServer := grpc.NewServer()
+	mockPeer := &mock.Server{GrpcServer: grpcServer, Listener: lis}
+	go mockPeer.Start()
+	return mockPeer, lis.Addr().String()
+}
+
+func PrepareClient(signed, processed chan *Elements, done chan struct{}, logger *log.Logger, threshold int, addr string) {
+	peer := Node{
+		Addr: addr,
+	}
+	Proposer, _ := CreateProposer(peer, logger)
+	go Proposer.Start(signed, processed, done, threshold)
+}
+
+func GetData() *Elements {
+	return &Elements{
+		SignedProp: &peer.SignedProposal{
+			ProposalBytes: []byte("signed-proposal-bytes"),
+			Signature:     []byte("signature"),
+		},
+	}
+}
```

Comment:
This it not necessary, you could simply inline with `&Elements{SignedProp: &peer.SignedProposal{}}`